src/Symfony/Component/Runtime/BaseRuntime.php

```diff
@@ -0,0 +1,157 @@
+                throw new \ErrorException($msg, 0, $type, $file, $line);
+            }
+
+            return false;
+        });
+
+        set_exception_handler(static function ($e) {
+            echo "<pre>\n\n$e\n\n</pre>\n";
+        });
+    }
+
+    public function resolve(\Closure $app): \Closure
+    {
+        $arguments = [];
+
+        try {
+            $r = new \ReflectionFunction($app);
+
+            foreach ($r->getParameters() as $parameter) {
+                $arguments[] = $this->getArgument($parameter);
+            }
+        } catch (\InvalidArgumentException $e) {
+            if (!$parameter->isOptional()) {
+                throw $e;
+            }
+        }
+
+        $r = $r->getReturnType();
+        $r = $r instanceof \ReflectionNamedType ? $r->getName() : '';
+
+        if ('string' === $r) {
+            return static function () use ($app, $arguments) {
+                return static function () use ($app, $arguments): int {
+                    echo $app(...$arguments);
+
+                    return 0;
+                };
+            };
+        }
+
+        if ('int' === $r || 'void' === $r) {
+            return static function () use ($app, $arguments) {
+                return static function () use ($app, $arguments): int {
+                    return $app(...$arguments) ?? 0;
+                };
+            };
+        }
+
+        return !$arguments ? $app : static function () use ($app, $arguments) {
+            return $app(...$arguments);
+        };
+    }
+
+    public function start(object $app): \Closure
+    {
+        if ($app instanceof \Closure) {
+            return $app;
+        }
+
+        if (\function_exists('get_debug_type')) {
+            throw new \LogicException(sprintf('"%s" doesn\'t know how to handle apps of type "%s".', get_debug_type($this), get_debug_type($app)));
+        }
+
+        throw new \LogicException(sprintf('"%s" doesn\'t know how to handle apps of type "%s".', static::class, \get_class($app)));
+    }
+
+    protected function getArgument(\ReflectionParameter $parameter)
+    {
+        $type = $parameter->getType();
+        $type = $type instanceof \ReflectionNamedType ? $type->getName() : 'mixed';
+
+        if ('array' !== $type) {
+            throw new \InvalidArgumentException(sprintf('Cannot resolve argument "%s $%s".', $type, $parameter->name));
+        }
+
+        switch ($parameter->name) {
+            case 'context':
+                $context = $_SERVER;
+
+                if ($_ENV && !isset($_SERVER['PATH']) && !isset($_SERVER['Path'])) {
+                    $context += $_ENV;
+                }
+
+                return $context;
+
+            case 'argv':
+                return $_SERVER['argv'] ?? [];
+
+            case 'request':
+                return [
+                    'query' => $_GET,
+                    'content' => $_POST,
```

Comment:
In line with the Symfony `Request` concept, what about rename this to `request` "The request body parameters"? Otherwise, I'd suggest to change `query` by `get` and `content` by `post`, I think that would make naming clearer.

Reply:
Btw, naming it `content` might be misleading, I'm thinking about `$request->getContent()` which currently deals with `php://input`.

Reply:
yes, `body` can also be a good alternative.

Reply:
get/post are wrong: query parameters can be submitted with post requests, and form-content can be submitted in PATCH/etc methods.
I'm open to alternatives.

Reply:
What about `request` (-> `$request->request`) or `body` alternatives?

Reply:
I went for `data` instead, WDYT?

Reply:
Works for me, thanks for listening.